core/templates/services/exploration-improvements-task-registry.service.ts

```diff
@@ -409,18 +430,49 @@ export class ExplorationImprovementsTaskRegistryService {
           stateName)
       ]),
     ]);
-    const supportingStateStats = new SupportingStateStats(
-      answerStats, cstPlaythroughIssues, eqPlaythroughIssues,
-      misPlaythroughIssues);
-    const newStateTasks = new StateTasks(tasksByType, supportingStateStats);
+    const supportingStats = new SupportingStateStats(
+      this.expStats.getStateStats(stateName), answerStats, cstPlaythroughIssues,
+      eqPlaythroughIssues, misPlaythroughIssues);
+    const newStateTasks = new StateTasks(tasksByType, supportingStats);
 
     for (const task of newStateTasks) {
-      this.tasksByType.get(task.taskType).push(task);
+      if (task.isOpen()) {
+        this.pushOpenTask(task);
+      }
     }
 
     this.tasksByState.set(stateName, newStateTasks);
     return newStateTasks;
   }
+
+  private refreshStateTasks(stateName: string): void {
```

Comment:
Prefix the names of private functions with an underscore.

Reply:
Sorry, this was my bad. I was misremembering an email thread, in which I thought we had made this decision, but it turns out we haven't. I'll bring this up at one of the team lead meetings since there have been some questions about the convention to use, but for now let's leave it as you have it!